src/expr/dtype.h

```diff
@@ -425,6 +425,12 @@ class DType
   const std::vector<std::shared_ptr<DTypeConstructor> >& getConstructors()
       const;
 
+  /**
+   * Return the subfield types of this datatype. This is the set of all types T
+   * for which there exists an argument to a constructor of type T.
+   */
+  std::unordered_set<TypeNode> getSubfieldTypes() const;
```

Comment:
With this function, do we still need `cvc5::theory::quantifiers::SygusTypeInfo::getSubfieldTypes`?

Reply:
There are 2 reasons:  `cvc5::theory::quantifiers::SygusTypeInfo::getSubfieldTypes` is (1) cached for efficiency, and (2) is slightly different: it is recursive and only follows "sygus" datatypes.

Reply:
Fair enough!